### DISCUSSION TITLE
Increment decrement in bulk?

### DISCUSSION THREAD

**Discussion by uzair004** [sequelize/sequelize#14372] [Help & Questions] [answered]
I want to update bulk of rows using sequelize , as there is not standard `bulkUpdate` method , I am using `bulkCreate` to find & update multiple instances.

**create a formatted array to insert/update into**
```
variants.forEach(() => {
   variantsWithUpdatedInventory.push({
          id: match.id,
          fk_item_varaint: match.fk_item_varaint,
          fk_kitchen_id: match.fk_kitchen_id,
          inventory: variant.closing,
          par_level: match.par_level,
          createdAt: moment(match.createdAt).unix,
          updatedAt: moment(match.updatedAt).unix,
        });
});
```

**Update inventory & updatedAt cols using bulkCreate**
```
// update foundVariants inventories
      await KitchenItemVariant.bulkCreate(variantsWithUpdatedInventory, {
        fields: ["id", "inventory", "updatedAt"],
        updateOnDuplicate: ["inventory", "updatedAt"],
        ignoreDuplicates: true,
        transaction,
      });
```

it was working fine, replacing inventory by new value. 

**Now I want to incremet decrement the existing inventory found in db instead of replacing it with new one** ??

**Comment by ephys** [comment, marked as answer]
There is a bulk update method, it's the static `update` method (as opposed to the instance `update` method): https://sequelize.org/api/v7/classes/model#update-1

Bulk increment / decrement are also available as static methods: https://sequelize.org/api/v7/classes/model#increment-1

**Reply by uzair004** [reply]
Are they available in v5 ?
I think that update changes single property/fields.  I want to update different rows with different inc and dec depending on calculations

**Reply by ephys** [reply]
I think so (though you should really upgrade to v6, EOL doesn't receive any security updates anymore)

You can use `literal()` to update each row based on the result of a calculation:

```javascript
MyModel.update({
  myAttribute: literal('some_column + 1')
}, {
  where: ...
});
```

That's what `Model.increment` and `Model.decrement` do
